tests/e2e/scale/test_scale_12_OCS_worker_nodes_and_6000_PVCs.py

```diff
@@ -0,0 +1,264 @@
+import logging
+import pytest
+import os
+
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility import utils, templating
+from ocs_ci.ocs import constants, scale_lib, machine
+from ocs_ci.utility.utils import ocsci_log_path
+from ocs_ci.ocs.scale_lib import FioPodScale
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.helpers import disruption_helpers
+from ocs_ci.ocs.node import get_worker_nodes
+from ocs_ci.framework.testlib import scale, E2ETest, ignore_leftovers
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_aws_i3,
+    skipif_bm,
+    skipif_external_mode,
+)
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+
+logger = logging.getLogger(__name__)
+
+# Scale data file
+log_path = ocsci_log_path()
+SCALE_DATA_FILE = f"{log_path}/scale_data_file.yaml"
+
+
+@ignore_leftovers
+@scale
+@skipif_aws_i3
+@skipif_bm
+@skipif_external_mode
+class TestAddNode(E2ETest):
+    """
+    Automates adding worker nodes to the cluster while IOs
+    """
+
+    skip_all = False
+
+    @pytest.mark.skipif("TestAddNode.skip_all")
+    def test_scale_node_and_capacity(self):
+        """
+        Test for scaling 12 OCS worker nodes to the cluster
+        Scale 12*3 = 36 OSDs
+        """
+
+        expected_worker_count = 12
+        osds_per_node = 3
+
+        try:
+            # Gather existing deviceset, OSD and node count in setup
+            existing_ocs_worker_list = get_worker_nodes()
+            existing_deviceset_count = storage_cluster.get_deviceset_count()
+            osd_replication_count = storage_cluster.get_osd_replica_count()
+            expected_deviceset_count = (
+                expected_worker_count / osds_per_node
+            ) * osd_replication_count
+
+            # Check existing OCS worker node count and add nodes if required
+            if len(existing_ocs_worker_list) < expected_worker_count:
+                scale_worker_count = expected_worker_count - len(
+                    existing_ocs_worker_list
+                )
+                scale_lib.scale_ocs_node(node_count=scale_worker_count)
+
+            # Check existing OSD count and add OSDs if required
+            if existing_deviceset_count < expected_deviceset_count:
+                add_deviceset_count = (
+                    expected_deviceset_count - existing_deviceset_count
```

Comment:
In LSO based cluster the the replica count is 1, does these lines of code take that also into factor while calculating the add_deviceset_count

Reply:
@PrasadDesala That's good point, let me add skipif_lso marker, anyway we don't have support to run some scale tests(scale capacity, node tests) in UPI cluster.